src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ConsoleDebugTargetsProvider.cs

```diff
@@ -321,24 +316,103 @@ public string GetFullPathOfExeFromEnvironmentPath(string exeToSearchFor)
                 return null;
             }
 
-            var paths = pathEnv.Split(new char[] {';'}, StringSplitOptions.RemoveEmptyEntries);
+            var paths = pathEnv.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
             foreach (var path in paths)
             {
                 // We don't want one bad path entry to derail the search
                 try
                 {
                     string exePath = Path.Combine(path, exeToSearchFor);
-                    if(TheFileSystem.FileExists(exePath))
+                    if (TheFileSystem.FileExists(exePath))
                     {
                         return exePath;
                     }
                 }
-                catch(ArgumentException)
+                catch (ArgumentException)
                 {
                 }
             }
 
             return null;
         }
+
+        /// <summary>
+        /// Escapes the given characters in a given string, ignoring escape sequences when inside a quoted string.
+        /// </summary>
+        /// <param name="unescaped">The string to escape.</param>
+        /// <param name="toEscape">The characters to escape in the string.</param>
+        /// <returns>The escaped string.</returns>
+        internal static string EscapeString(string unescaped, char[] toEscape)
+        {
+            if (string.IsNullOrWhiteSpace(unescaped)) return unescaped;
+
+            var currentState = StringState.NoSlash;
+            var finalBuilder = new StringBuilder();
+            foreach (var currentLetter in unescaped)
+            {
+                switch (currentState)
+                {
+                    case StringState.NoSlash:
+                        // If we're currently not in a quoted string, then we need to escape anything in toEscape.
+                        // The valid transitions are to OneSlash and InString. Any '"' encountered here is a raw quote.
+                        if (currentLetter == '\\')
+                        {
+                            currentState = StringState.OneSlash;
+                        }
+                        else if (currentLetter == '"')
+                        {
+                            currentState = StringState.InString;
+                        }
+                        else if (toEscape.Contains(currentLetter))
+                        {
+                            finalBuilder.Append('^');
+                        }
+
+                        finalBuilder.Append(currentLetter);
+                        break;
+                    case StringState.OneSlash:
```

Comment:
The only thing I'm a little wondering about is whether or not this is actually a valid state. Does `\"` have special meaning in the shell? A quick test of `echo \"` prints `\"`, so I think this is actually incorrect.